app/presenters/post_presenter.rb

```diff
@@ -47,6 +47,10 @@ def page_title
     post_page_title @post
   end
 
+  def poll_participation_answer_id
+    @post.poll.participation_answer(current_user).poll_answer_id if already_participated_in_poll
```

Comment:
You can combine this with `already_participated_in_poll`, so this line looks like this:
```ruby
@post.poll&.participation_answer(current_user)&.poll_answer_id if user_signed_in?
```
and then also use `poll_participation_answer_id` to decide "if already participated" in the `poll_view.js`.

Then you can remove `already_participated_in_poll` from `post_presenter.rb` and remove `already_participated?` from `poll.rb`.

This has two reasons:
1. less code
2. less database queries

with the old way you queried the participation 3 times from the database:
1. for the `if already_participated_in_poll` in this line
2. for the `poll.participation_answer(current_user)` here
3. again `already_participated_in_poll` to add it to `non_directly_retrieved_attributes`.

With the new line you do only one query, because you can decide "if already participated" if this method returns nil or the id.